software/bsg_manycore_lib/bsg_striped_array.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+extern "C" {
+#include "bsg_manycore.h"
+}
+
+namespace bsg_manycore {
+    template <typename TYPE, int SIZE, int TG_SIZE_X, int TG_SIZE_Y, int STRIPE_SIZE=1>
```

Comment:
The template parameters should probably be `size_t` (except for TYPE). Also what is STRIPE_SIZE?

Reply:
It's if you want more than one consecutive element on the current tile. And sure.

Reply:
Oh that makes sense.

Reply:
Is stripe size the right name for this? Size implies bytes, but nelems (or something) would imply that more than one element is used

Ditto for TG_SIZE_X/Y, and SIZE. Why not TG_DIM_X and TG_DIM_Y?

Reply:
Is stripe size the right name for this? Size implies bytes, but nelems (or something) would imply that more than one element is used

Ditto for TG_SIZE_X/Y, and SIZE. Why not TG_DIM_X and TG_DIM_Y?

Reply:
I'm not really sure I agree that SIZE implies bytes in this case because that wouldn't really make any sense at all. It should be noted that many standard C++ containers use the size() function to mean the number of elements, not the number of bytes, in the data structure.

As for TG_SIZE_X/Y vs TG_DIM_X/Y - sure let's change the name.